zone/spell_effects.cpp

```diff
@@ -1445,35 +1445,104 @@ bool Mob::SpellEffect(Mob* caster, uint16 spell_id, float partial, int level_ove
 				}
 				// Racial Illusions
 				else {
-					SendIllusionPacket
-					(
-						spell.base[i],
-						Mob::GetDefaultGender(spell.base[i], GetGender()),
-						spell.base2[i],
-						spell.max[i]
-					);
-					if(spell.base[i] == OGRE){
-						SendAppearancePacket(AT_Size, 9);
+					if (spell.max[i] > 0) {
+						if (spell.base2[i] == 0) {
+							SendIllusionPacket(
+								spell.base[i],
+								(spell.max[i] - 1)
+							);
+						} else {
+							if (spell.max[i] != 3) {
+								SendIllusionPacket(
+									spell.base[i],
+									(spell.max[i] - 1),
+									spell.base2[i],
+									spell.max[i]
+								);
+							} else {
+								SendIllusionPacket(
+									spell.base[i],
+									Mob::GetDefaultGender(spell.base[i], GetGender()),
+									spell.base2[i],
+									spell.base2[i]
+								);
+							}
+						}
+					} else {
+						SendIllusionPacket(
+							spell.base[i],
+							Mob::GetDefaultGender(spell.base[i], GetGender()),
+							spell.base2[i],
+							spell.max[i]
+						);
 					}
-					else if(spell.base[i] == TROLL){
```

Comment:
Can this size shit be replaced with https://github.com/EQEmu/Server/blob/31ab1d42874fbc39d98c05d5d62096fb0f8610d9/common/races.cpp#L1488 ?

Reply:
Yes. Didn't know that was a thing.